flink-tests/src/test/java/org/apache/flink/test/checkpointing/ContinuousFileProcessingCheckpointITCase.java

```diff
@@ -217,13 +217,12 @@ public void invoke(String value) throws Exception {
 			}
 
 			// add some latency so that we have at least one checkpoint in
-			if (!hasFailed && successfulCheckpoints == 0) {
-				Thread.sleep(2);
+			if (!hasFailed && !hasSuccessfulCheckpoints) {
+				Thread.sleep(5);
```

Comment:
The problem is that sleeping never releases the checkpoint lock (which is held when the invoke is called), so the checkpoint will never pass no matter how much we sleep. And we do not have access to the checkpoint lock from within the invoke. This is why we just sleep for 5 (with the checkpoint interval set to 10). This way, with high probability, we will have a checkpoint in the first couple of elements.